File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    number += 1
    dp = [0 for _ in range(number)]
    
    for i in range(1, number):
        for j in range(i, number, i):
            dp[j] += 1
    
    return sum(power if dp[i] > limit else dp[i] for i in range(number))

File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.cs
public class Solution {
    public int solution(int number, int limit, int power) {
        number += 1;
        int[] dp = new int[number];
        
        for (int i = 1; i < number; i++)
        {
            for (int j = i; j < number; j += i)
            {
                dp[j] += 1;
            }
        }
        
        int answer = 0;
        
        foreach (int num in dp)
        {
            answer += (num > limit)? power : num;
        }
        
        return answer;
    }
}